velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -36,4 +38,101 @@ struct YearFunction : public InitSessionTimezone<T> {
     result = getYear(getDateTime(date));
   }
 };
+
+template <typename T>
+struct UnixTimestampFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  // unix_timestamp();
+  // If no parameters, return the current unix timestamp.
+
+  FOLLY_ALWAYS_INLINE void call(int64_t& result) {
+    result = Timestamp::now().getSeconds();
+  }
+
+  // unix_timestamp(input);
+  // If format is not specified, assume kDefaultFormat.
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const core::QueryConfig& config,
+      const arg_type<Varchar>* /*input*/) {
+    format_ = buildJodaDateTimeFormatter(kDefaultFormat_);
+    isConstFormat_ = true;
+    setTimezone(config);
+  }
+
+  FOLLY_ALWAYS_INLINE bool call(
+      int64_t& result,
+      const arg_type<Varchar>& input) {
+    DateTimeResult dateTimeResult;
+    try {
+      dateTimeResult =
+          format_->parse(std::string_view(input.data(), input.size()));
+    } catch (const VeloxUserError&) {
+      // Return null if could not parse.
+      return false;
+    }
+    dateTimeResult.timestamp.toGMT(getTimezoneId(dateTimeResult));
+    result = dateTimeResult.timestamp.getSeconds();
+    return true;
+  }
+
+  // unix_timestamp(input, format):
+  // If format is constant, compile it just once per batch.
+
+  FOLLY_ALWAYS_INLINE void initialize(
+      const core::QueryConfig& config,
+      const arg_type<Varchar>* /*input*/,
+      const arg_type<Varchar>* format) {
+    if (format != nullptr) {
+      format_ = buildJodaDateTimeFormatter(
```

Comment:
Should we handle errors here? Looks like the semantic is to return null in case of an error. Let's add some test cases.